source/api_cc/src/DeepPot.cc

```diff
@@ -1269,188 +1256,17 @@ void DeepPotModelDevi::init(const std::vector<std::string>& models,
     return;
   }
   numb_models = models.size();
-  sessions.resize(numb_models);
-  graph_defs.resize(numb_models);
-
-  int gpu_num = -1;
-#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
-  DPGetDeviceCount(gpu_num);
-#endif  // GOOGLE_CUDA || TENSORFLOW_USE_ROCM
-
-  SessionOptions options;
-  get_env_nthreads(num_intra_nthreads, num_inter_nthreads);
-  options.config.set_inter_op_parallelism_threads(num_inter_nthreads);
-  options.config.set_intra_op_parallelism_threads(num_intra_nthreads);
-  for (unsigned ii = 0; ii < numb_models; ++ii) {
-    graph_defs[ii] = new GraphDef();
-    if (file_contents.size() == 0) {
-      check_status(ReadBinaryProto(Env::Default(), models[ii], graph_defs[ii]));
-    } else {
-      (*graph_defs[ii]).ParseFromString(file_contents[ii]);
-    }
-  }
-#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
-  if (gpu_num > 0) {
-    options.config.set_allow_soft_placement(true);
-    options.config.mutable_gpu_options()->set_per_process_gpu_memory_fraction(
-        0.9);
-    options.config.mutable_gpu_options()->set_allow_growth(true);
-    DPErrcheck(DPSetDevice(gpu_rank % gpu_num));
-  }
-#endif  // GOOGLE_CUDA || TENSORFLOW_USE_ROCM
-
-  for (unsigned ii = 0; ii < numb_models; ++ii) {
-    if (gpu_num > 0) {
-      std::string str = "/gpu:";
-      str += std::to_string(gpu_rank % gpu_num);
-      graph::SetDefaultDevice(str, &(*graph_defs[ii]));
-    }
-    check_status(NewSession(options, &(sessions[ii])));
-    check_status(sessions[ii]->Create(*graph_defs[ii]));
-  }
-  try {
-    model_version = get_scalar<STRINGTYPE>("model_attr/model_version");
-  } catch (deepmd::tf_exception& e) {
-    // no model version defined in old models
-    model_version = "0.0";
-  }
-  if (!model_compatable(model_version)) {
-    throw deepmd::deepmd_exception(
-        "incompatable model: version " + model_version +
-        " in graph, but version " + global_model_version +
-        " supported. "
-        "See https://deepmd.rtfd.io/compatability/ for details.");
-  }
-  dtype = session_get_dtype(sessions[0], "descrpt_attr/rcut");
-  if (dtype == tensorflow::DT_DOUBLE) {
-    rcut = get_scalar<double>("descrpt_attr/rcut");
-  } else {
-    rcut = get_scalar<float>("descrpt_attr/rcut");
-  }
-  cell_size = rcut;
-  ntypes = get_scalar<int>("descrpt_attr/ntypes");
-  try {
-    ntypes_spin = get_scalar<int>("spin_attr/ntypes_spin");
-  } catch (const deepmd::deepmd_exception) {
-    ntypes_spin = 0;
-  }
-  dfparam = get_scalar<int>("fitting_attr/dfparam");
-  daparam = get_scalar<int>("fitting_attr/daparam");
-  if (dfparam < 0) {
-    dfparam = 0;
+  if (numb_models == 0) {
```

Comment:
Consider adding a check to ensure that the `models` vector is not empty before initializing the `dps` vector. This can prevent potential issues with empty models. [medium]

Reply:
This is what I have done?